### DISCUSSION TITLE
react on native crashes

### DISCUSSION THREAD

**Discussion by pke** [getsentry/sentry-react-native#2646] [Q&A]
I'd like to write a value to a settings store when the app crashes with a native exception. Is there a way to hook into Sentry's mechanism when it reports a native crash?

Or I would have to design my system differently and somehow detect an unclear app exit (via native crash) during the next app restart.

**Comment by krystofwoldrich** [comment]
Hi,
somehow this went under the radar, I'm sorry for that.

At the moment RN SDK doesn't have this functionality, but we are planning for it, see the linked issues. Native SDKs have this functionality, there is an example of how to use those if you are comfortable with Java/Obj-C.

- https://github.com/getsentry/sentry-react-native/issues/2797
- https://github.com/getsentry/sentry-react-native/issues/2789

**Reply by astanb** [reply]
Has this been implemented yet? My use case is persisting a large piece of state, but before the app is closed, that will then be reloaded on the next launch. If there was a way to hook into Sentry's native exception handler it would be very useful.
